routes/panel.js

```diff
@@ -559,6 +553,178 @@ router.post('/api/showPic', function(req, res) {
 	}
 });
 
+/*** Events ***/
+
+// Get events for logged in user
+router.get('/api/userevents', function(req, res) {
+	if (req.session.user == null) {
+		// not logged in, redirect to log in page
+		res.redirect('/panel');
+	}
+	else {
+		// return list of events belonging to current user
+		events.getEventsForUser(req.session.user.username, function(err, userEvents) {
+			if (err) { console.log("failed to retrieve events for user: ", err); }
+			else {
+				res.json(userEvents);
+			}
+		});
+	}
+});
+
+// Add new event
+router.post('/api/addEvent', function(req, res) {
+	if (req.session.user == null) {
+		// not logged in, redirect to log in page
+		res.redirect('/panel');
+	}
+	else {
+		events.addNewEvent(req.body.name, req.body.type, [req.session.user.username], function(err, saved) {
+			if (err) {
+				console.log("failed to add event for user: ", err); 
+				res.json({"success": false, "err": err});
+			}
+
+			else {
+				// return full list of user's events
+				res.redirect('/panel/api/userevents');
+			}
+		});
+	}
+});
+
+router.get('/api/eventData/:id', function(req, res) {
+	if (req.session.user == null) {
+		// not logged in, redirect to log in page
+		res.status(400).send();
+	}
+	else {
+		events.userHasAccessToEvent(req.session.user.username, req.params.id, function(hasAccess) {
+			if (!hasAccess) {
+				// user doesn't have access to this event
+				console.log("user requested event they don't have access to");
+				res.status(400).send();
+				return;
+			}
+			else {
+				events.getEventByID(req.params.id, function(err, o) {
+					if (o) {
+						res.json(o);
+					}
+					else {
+						res.status(400).send(err);
+					}
+				});
+			}
+		});
+	}
+});
+
+// update details for one event 
+router.post('/api/updateEvent', function(req, res) {
+	if (req.session.user == null) {
+		// not logged in, redirect to log in page
+		res.redirect('/panel');
+		console.log("not logged in ");
+	}
+	else {
+		console.log(req.body);
```

Comment:
All the console.logs in this file were essentially copied over from the shows api...should i still get rid of them? Should I get rid of the ones for shows too then?